lib/src/util/function_signature.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:tuple/tuple.dart';
+
+import '../ast/sass.dart';
+import '../exception.dart';
+import '../parse/scss.dart';
+import '../utils.dart';
+
+/// Parses a function signature of the format allowed by Node Sass's functions
+/// option and returns its name and declaration.
+///
+/// If [requireParens] is `false`, this allows parentheses to be omitted.
+///
+/// Throws a [SassFormatException] if parsing fails.
+Tuple2<String, ArgumentDeclaration> parseSignature(String signature,
```

Comment:
Nit: just include this in `lib/src/util.dart`

Reply:
In the current structure `lib/src/utils.dart` only have high level helper functions, and all feature specific helpers are in their own files in `lib/src/util/`. - This is why I created a separate file. @nex3 let me know if you still prefer to put this function in `lib/src/utils.dart`.

Reply:
@nex3 Any thoughts?

Reply:
Yeah, generally I prefer to only add sub-libraries for *groups* of utility functions. If there's only one function of a given type, it should go in the general utility library.